GooglePlayServicesLab/app/src/main/java/com/justinwells/myapplication/MainActivity.java

```diff
@@ -0,0 +1,174 @@
+import com.google.android.gms.maps.OnMapReadyCallback;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.MarkerOptions;
+
+import static com.justinwells.myapplication.R.id.map;
+
+public class MainActivity extends FragmentActivity implements OnMapReadyCallback, GoogleApiClient.OnConnectionFailedListener, GoogleApiClient.ConnectionCallbacks {
+
+    LocationRequest mLocationRequest;
+    public static final String KEY = "AIzaSyAzKiVxSpALnAUI_b4IEVYS9T6ivYlZpfU";
+    GoogleApiClient mGoogleApiClient;
+    Location mLocation;
+    MapFragment mMapFragment;
+    double mLatitude, mLongitude;
+    public static final int MY_PERMISSIONS_REQUEST_LOCATION = 138;
+    public static final String[] REQUEST_CODE_LOCATION = {
+            android.Manifest.permission.ACCESS_COARSE_LOCATION};
+    PendingResult<LocationSettingsResult> result;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+
+        mLocationRequest = new LocationRequest();
+        mLocationRequest.setInterval(10000);
+        mLocationRequest.setFastestInterval(5000);
+        mLocationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+
+        LocationSettingsRequest.Builder builder = new LocationSettingsRequest.Builder()
+                .addLocationRequest(mLocationRequest);
+
+
+        if (mGoogleApiClient == null) {
+            mGoogleApiClient = new GoogleApiClient.Builder(this)
+                    .addConnectionCallbacks(this)
+                    .addOnConnectionFailedListener(this)
+                    .addApi(LocationServices.API)
+                    .build();
+        }
+
+        result =
+                LocationServices.SettingsApi.checkLocationSettings(mGoogleApiClient,
+                        builder.build());
+
+        MapFragment mapFragment = (MapFragment) getFragmentManager()
+                .findFragmentById(map);
+        mapFragment.getMapAsync(this);
+
+
+    }
+
+    protected void onStart() {
+        mGoogleApiClient.connect();
+        super.onStart();
+    }
+
+    protected void onStop() {
+        mGoogleApiClient.disconnect();
+        super.onStop();
+    }
+
+
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+
+
+        googleMap.addMarker(new MarkerOptions()
+                .position(new LatLng(mLatitude, mLongitude))
+                .title("Marker"));
+
+        googleMap.moveCamera(CameraUpdateFactory.newLatLngZoom(new LatLng(mLatitude, mLongitude), 15));
+    }
+
+    @Override
+    public void onConnected(@Nullable Bundle bundle) {
+
+        if (ActivityCompat.checkSelfPermission(this, android.Manifest.permission.ACCESS_COARSE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
+            mLocation = LocationServices.FusedLocationApi.getLastLocation(
+                    mGoogleApiClient);
+            if (mLocation != null) {
```

Comment:
You might need a `LocationListener` if `getLastLocation()` returns null at first